test/regression-test.sh

```diff
@@ -157,18 +181,21 @@ grep -E '\s+-\w\s+' <<<"$*" > /dev/null && {
 #######################################
 
 mkdir -p "$CURRENT_OUTPUT_DIR"
-run_tests "$CURRENT_OUTPUT_DIR" 'current' "$@"
+run_tests "$CURRENT_OUTPUT_DIR" 'current' "${TEST_FILES[@]}"
 
 #######################################
 # Diffing
 #######################################
 
-basename -a "$@" | while read -r TEST_SCRIPT; do
+EXIT_CODE=0
+while read -r TEST_SCRIPT; do
     echo -ne "${YELLOW}[DIFF]${RESET} Compare ${YELLOW}$TEST_SCRIPT${RESET} "
     echo 'output with snapshot'
     diff -su --color=always \
         --label "$TEST_SCRIPT@snapshot" \
         --label "$TEST_SCRIPT@current" \
         "$CURRENT_OUTPUT_DIR/$TEST_SCRIPT.out" \
-        "$SNAPSHOTS_DIR/$TEST_SCRIPT.out"
-done
+        "$SNAPSHOTS_DIR/$TEST_SCRIPT.out" \
+        || EXIT_CODE=1
+done < <(basename -a "${TEST_FILES[@]}")
+exit "$EXIT_CODE"
```

Comment:
A cool little thing this script could do is to print instructions on what to do when it fails. Something like "Some test outputs were different from the snapshots. This means that either the tests failed or the snapshots are outdated. If you think the new test output is correct run `bash test/regression-test.sh -u` to update the snapshots".
I feel like this is particularly useful when run by pre-commit, so you might condition this message with a flag paased by `pre-commit`, but I don't think it hurts to always have it.